code/game/machinery/camera/trackable.dm

```diff
@@ -82,47 +61,140 @@
+
+	if(attempt_track())
+		set_rechecking(FALSE)
+		return
+
+	if(cameraticks < CAMERA_TICK_LIMIT)
+		if(!cameraticks)
+			to_chat(tracking_holder, span_warning("Target is not near any active cameras. Attempting to reacquire..."))
+		cameraticks++
+		return
 
-///Called by Signals, used to cancel tracking of a target.
-/datum/trackable/proc/cancel_target_tracking(atom/source)
+	to_chat(tracking_holder, span_warning("Unable to reacquire, cancelling track..."))
+	reset_tracking()
+
+/// Tries to track onto our target mob. Returns true if it succeeds, false otherwise
+/datum/trackable/proc/attempt_track()
+	if(!tracked_mob)
+		reset_tracking()
+		return FALSE
+
+	if(!tracked_mob.can_track(tracking_holder))
+		return FALSE
+	// In case we've been checking
+	set_rechecking(FALSE)
+	SEND_SIGNAL(src, COMSIG_TRACKABLE_TRACKING_TARGET, tracked_mob)
+	return TRUE
+
+/datum/trackable/proc/glide_size_changed(datum/source, new_glide_size)
 	SIGNAL_HANDLER
-	set_tracking(FALSE)
+	SEND_SIGNAL(src, COMSIG_TRACKABLE_GLIDE_CHANGED, tracked_mob, new_glide_size)
 
 /**
- * set_tracked_mob
+ * reset_tracking
  *
- * Sets a mob as being tracked, if a target is already provided then it will track that directly,
- * otherwise it will give a tgui input list to find targets to track.
+ * Resets our tracking
+ */
+/datum/trackable/proc/reset_tracking()
+	set_tracked_mob(null)
+
+/**
+ * track_input
+ *
+ * Sets a mob as being tracked, will give a tgui input list to find targets to track.
  * Args:
  *  tracker - The person trying to track, used for feedback messages. This is not the same as tracking_holder
- *  tracked_mob_name - (Optional) The person being tracked, to skip the input list.
  */
-/datum/trackable/proc/set_tracked_mob(mob/living/tracker, tracked_mob_name)
+/datum/trackable/proc/track_input(mob/living/tracker)
 	if(!tracker || tracker.stat == DEAD)
 		return
 
-	if(tracked_mob_name)
-		find_trackable_mobs() //this is in case the tracked mob is newly/no-longer in camera field of view.
-		tracked_mob = isnull(humans[tracked_mob_name]) ? others[tracked_mob_name] : humans[tracked_mob_name]
-		if(isnull(tracked_mob))
-			to_chat(tracker, span_notice("Target is not on or near any active cameras. Tracking failed."))
-			return
-		to_chat(tracker, span_notice("Now tracking [tracked_mob_name] on camera."))
-	else
-		var/target_name = tgui_input_list(tracker, "Select a target", "Tracking", find_trackable_mobs())
-		if(!target_name || isnull(target_name))
-			return
-		tracked_mob = isnull(humans[target_name]) ? others[target_name] : humans[target_name]
+	var/target_name = tgui_input_list(tracker, "Select a target", "Tracking", find_trackable_mobs())
+	if(!target_name || isnull(target_name))
+		return
+	var/datum/weakref/mob_ref = isnull(humans[target_name]) ? others[target_name] : humans[target_name]
+	if(isnull(mob_ref))
+		to_chat(tracker, span_notice("Target is not on or near any active cameras. Tracking failed."))
+		return
+	set_tracked_mob(mob_ref?.resolve())
```

Comment:
why do we need the `?` if we already do an explicit `isnull()` check?

Reply:
I am a fool